Frontend/ui-customer/src/Flow.purs

```diff
@@ -310,8 +313,22 @@ currentFlowStatus = do
       else do
         resp <- lift $ lift $ Remote.updateProfile (Remote.mkUpdateProfileRequest)
         pure unit
+      void $ pure $ setCleverTapUserData "Name" ((fromMaybe "" response.firstName) <> " " <> (fromMaybe "" response.middleName) <> " " <> (fromMaybe "" response.lastName))
+
+      if (fromMaybe "UNKNOWN" (response.gender) /= "UNKNOWN") then do 
+        void $ pure $ setCleverTapUserData "gender" (fromMaybe "" response.gender)
+        pure unit
+        else pure unit
+      if (isJust response.language) then do
+        void $ pure $ setCleverTapUserProp "Preferred Language" (fromMaybe "" response.language)
+        pure unit
+        else pure unit
```

Comment:
use when when `else`is not required, also you can remove `pure unit` if already using void

Reply:
other places as well